jax/_src/nn/functions.py

```diff
@@ -765,3 +767,172 @@ def hard_silu(x: ArrayLike) -> Array:
   return x_arr * hard_sigmoid(x_arr)
 
 hard_swish = hard_silu
+
+def _get_large_negative(dtype):
+  dtype_max = jnp.finfo(dtype).max
+  return jnp.asarray(-0.7 * dtype_max, dtype=dtype)
```

Comment:
Can you add a comment explaining the magic number here? Why is -0.7 used?

Reply:
Emm, actually, I am not sure about the rational of using -0.7. But it seems to be a rule of thumb used in various places: [praxis](https://github.com/google/praxis/blob/c41477c601fea125ae58f136f139758c34d121b8/praxis/py_utils.py#L729), 
[jax paged attention](https://github.com/google/jax/blob/a4c92a454b23e2d10a49d28fe0fe39514ec558c8/tests/pallas/paged_attention_kernel_test.py#L94), [jax splash attention](https://github.com/google/jax/blob/a4c92a454b23e2d10a49d28fe0fe39514ec558c8/jax/experimental/pallas/ops/tpu/splash_attention/splash_attention_kernel.py#L37), etc.

Reply:
I cannot find explanation from these places.